src/app/src/components/FilterSidebar.jsx

```diff
@@ -0,0 +1,178 @@
+import React, { Component } from 'react';
+import { func, oneOf } from 'prop-types';
+import { connect } from 'react-redux';
+import AppBar from '@material-ui/core/AppBar';
+import Tabs from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
+import Typography from '@material-ui/core/Typography';
+
+import FilterSidebarGuideTab from './FilterSidebarGuideTab';
+import FilterSidebarSearchTab from './FilterSidebarSearchTab';
+
+import {
+    filterSidebarTabsEnum,
+    filterSidebarTabs,
+} from '../util/constants';
+
+import {
+    makeSidebarGuideTabActive,
+    makeSidebarSearchTabActive,
+} from '../actions/ui';
+
+import {
+    fetchContributorOptions,
+    fetchContributorTypeOptions,
+    fetchCountryOptions,
+    fetchAllFilterOptions,
+} from '../actions/filterOptions';
+
+import {
+    contributorOptionsPropType,
+    contributorTypeOptionsPropType,
+    countryOptionsPropType,
+} from '../util/propTypes';
+
+import { allListsAreEmpty } from '../util/util';
+
+class FilterSidebar extends Component {
+    componentDidMount() {
+        const {
+            contributorsData,
+            contributorTypesData,
+            countriesData,
+            fetchFilterOptions,
+            fetchContributors,
+            fetchContributorTypes,
+            fetchCountries,
+        } = this.props;
+
+        if (allListsAreEmpty(contributorsData, contributorTypesData, countriesData)) {
+            return fetchFilterOptions();
+        }
+
+        if (!contributorsData.length) {
```

Comment:
Can you help me understand why we need to have these individual fetch calls? I think that we will always be fetching all the options, so `fetchFilterOptions` looks like all we need. It's good that we are dispatching three separate actions so we can fetch them in parallel.

Reply:
Sure -- so this all happens in `componentDidMount` for this component. It's possible and likely that this specific component will be unmounted for a variety of reasons in the normal use of the app: for instance, I land on the main page, the options get fetched, then I go to the login page, then return to the main page.

On returning to the main page after login, I thought it'd be desirable not to have to re-fetch all these options again if they were already available in the Redux state. However, it's possible that 2/3 succeed and the other fails for whatever reason, so in that case I set this up so that the component will fetch options for any of them that are empty arrays in the Redux state in `componentDidMount`.

In practice I think it'll typically be the case that all of them or none of them succeed, but I can imagine the possibility that one of them succeeds and the others don't.

That said I can set this up to make it so that we `fetchAllOptions` unconditionally every time this component mounts and reset all options every time it unmounts. That would cause each client to make 3 get requests each time this component re-mounts, but that may not be too problematic.

Reply:
Thanks for the explanation. It's sound reasoning and it makes sense to leave it as is.